azure-slurm/slurmcc/topology.py

```diff
@@ -6,29 +6,53 @@
 import subprocess as subprocesslib
 import datetime
 from . import util as slutil
+from enum import Enum
 
 log=logging.getLogger('topology')
-
+class TopologyType(Enum):
+    """
+    Enum class representing the type of topology.
+    """
+    BLOCK = "block"
+    TREE = "tree"
+    def __str__(self):
+        return self.value
+class TopologyInput(Enum):
+    """
+    Enum class representing the type of topology input.
+    """
+    NVLINK = "nvlink"
+    FABRIC = "fabric"
+    def __str__(self):
+        return self.value
 class Topology:
     """
     A class to represent and manage the topology of a Slurm cluster.
     Attributes:
     """
 
-    def __init__(self,partition, output,directory):
+    def __init__(self,partition,output,topo_input,topo_type,directory):
```

Comment:
just for readability use more whitespace (space after commas, and blank line between method defs)